### PR TITLE
Add Eloquent perfomance tests

### PR DESCRIPTION
**Pull request by ahcorde**
I'm going to open some pull request to start discussing and merging upstream the performance tests that @cottsay has developed.

Potentials TODO task:
 - This PR includes 3 new jobs, one of them with prerequisites of the test and other one just with the tests. Maybe it makes sense to include everything in the same job.
 - These performance tests need two repos defined [here](https://gist.githubusercontent.com/cottsay/d3ed9d3c7ed32165fdb987a09d434661/raw/ros2.repos)
   - [buildfarm_perf_tests]( https://github.com/cottsay/buildfarm_perf_tests.git) This repo contains the tests. With the results of these tests we will create the plots.
       - **Where we should locate this repo?**
               - Should we create on new repo in osrf?
               - Should we try to merge this in Apex.AI performance_test repo?
   - [performance_test](https://gitlab.com/cottsay/performance_test.git). Right now @cottsay has it own fork because there are two pending PRs:
                - https://gitlab.com/ApexAI/performance_test/merge_requests/121
                - https://gitlab.com/ApexAI/performance_test/merge_requests/129

Related PR https://github.com/ros-infrastructure/ros_buildfarm/pull/689

## Jenkings Plugin
 - [plot-plugin](https://github.com/jenkinsci/plot-plugin)
 - [image-gallery-plugin](https://github.com/jenkinsci/image-gallery-plugin)

## Documentation
 - [How to run the tests locally](https://github.com/cottsay/buildfarm_perf_tests/pull/2)

### CODE REVIEW

File: eloquent/ci-nightly-performance-prerequisites.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- ros-melodic-rospy-tutorials
+- ros-melodic-tf2-msgs
+jenkins_job_label: buildagent_arm64
+jenkins_job_priority: 50
+jenkins_job_schedule: 15 23 * * *
+jenkins_job_timeout: 300
+package_selection_args: '--packages-up-to buildfarm_perf_tests --packages-skip performance_test buildfarm_perf_tests --packages-skip-regex .*connext.*'
+repos_files:
+- https://github.com/ros2/ros2/raw/master/ros2.repos
+- https://gist.githubusercontent.com/cottsay/d3ed9d3c7ed32165fdb987a09d434661/raw/ros2.repos
```
Comment (chapulina) [inline eloquent/ci-nightly-performance-prerequisites.yaml]: I'm not sure what our policy is about this, maybe someone else from the team has an opinion on what to do here?

My suggestion would be to move @cottsay's changes to https://github.com/ros2/performance_test and use that until everything has been merged upstream into Apex's repo.

**Review by chapulina — COMMENTED** [review]
> buildfarm_perf_tests This repo contains the tests... Where we should locate this repo? ... new repo in osrf? ... merge this in Apex.AI performance_test repo?

Since the other PRs to Apex's `performance_test` are taking a while to get in, I'd suggest we just host that bit separately for now. I'm not sure whether the `ros2` org or `ros-infrastructure` would be the best place. Maybe @dirk-thomas or @nuclearsandwich have a preference?

File: eloquent/ci-nightly-performance-prerequisites.yaml (hunk 1 above)
Comment (cottsay) [reply eloquent/ci-nightly-performance-prerequisites.yaml]: I agree, let's fork the repo there until the upstream changes are merged. It is easy to switch later.

**Comment by cottsay** [comment]
> I'm not sure whether the ros2 org or ros-infrastructure would be the best place.

I'd rather see that repository move under `ros2` than `ros-infrastructure`. We could also host the `.repos` file inside of that repo instead of in a Gist, similar to what we do for the navigation2 jobs.

File: eloquent/ci-nightly-performance-prerequisites.yaml (hunk 1 above)
Comment (cottsay) [reply eloquent/ci-nightly-performance-prerequisites.yaml]: Both of the repos are now under the `ros2` org. Hopefully Apex will merge our two PRs soon, which should eliminate one of them.

If we can get the packages listed in the official distribution, we can eliminate the repos file and just list the packages, which would be a lot cleaner.

**Comment by ahcorde** [comment]
closing this PR becuase of this other one https://github.com/ros2/ros_buildfarm_config/pull/53

**Pull request closed without merging**